src/python_testing/TC_DISHM_1_2.py

```diff
@@ -0,0 +1,114 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_DISHM_1_2(MatterBaseTest):
+
+    async def read_mode_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.DishwasherMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    @async_test_body
+    async def test_TC_DISHM_1_2(self):
+
+        self.endpoint = self.user_params.get("endpoint", 1)
+        logging.info("This test expects to find this cluster on endpoint 1")
+
+        asserts.assert_true(self.check_pics("DISHM.S.A0000"), "DISHM.S.A0000 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.A0001"), "DISHM.S.A0001 must be supported")
+
+        attributes = Clusters.DishwasherMode.Attributes
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "Read SupportedModes attribute")
+        supported_modes = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
```

Comment:
There is no `read_mod_attribute_expect_success` function on `self`.... It should be `read_mode_attribute_expect_success`.  This applies in a few places below too.